django/db/models/fields/__init__.py

```diff
@@ -1725,54 +1725,84 @@ def check(self, **kwargs):
         return errors
 
     def _check_decimal_places(self):
-        try:
-            decimal_places = int(self.decimal_places)
-            if decimal_places < 0:
-                raise ValueError()
-        except TypeError:
-            return [
-                checks.Error(
-                    "DecimalFields must define a 'decimal_places' attribute.",
-                    obj=self,
-                    id="fields.E130",
-                )
-            ]
-        except ValueError:
-            return [
-                checks.Error(
-                    "'decimal_places' must be a non-negative integer.",
-                    obj=self,
-                    id="fields.E131",
-                )
-            ]
+        if self.decimal_places is None:
+            if not (
+                connection.features.supports_no_precision_decimalfield
+                or "supports_no_precision_decimalfield"
+                in self.model._meta.required_db_features
+            ):
```

Comment:
I think it's a bit clearer to distribute the `not` here:

```suggestion
            if (
                not connection.features.supports_no_precision_decimalfield
                and "supports_no_precision_decimalfield"
                not in self.model._meta.required_db_features
            ):
```